model_api/python/model_api/tilers/detection.py

```diff
@@ -49,26 +49,21 @@ def parameters(cls):
         )
         return parameters
 
-    def _postprocess_tile(self, predictions, coord):
+    def _postprocess_tile(
+        self,
+        predictions: DetectionResult,
+        coord: list[int],
+    ) -> dict:
         """Converts predictions to a format convenient for further merging.
 
         Args:
-             predictions: predictions from a detection model: a list of `Detection` objects
-             or one `DetectionResult`
-             coord: a list containing coordinates for the processed tile
+            predictions: predictions wrapped in DetectionResult from a detection model
+            coord: a list containing coordinates for the processed tile
 
         Returns:
-             a dict with postprocessed predictions in 6-items format: (label id, score, bbox)
+            a dict with postprocessed predictions in 6-items format: (label id, score, bbox)
         """
         output_dict = {}
-        if hasattr(predictions, "objects"):
-            detections = _detection2array(predictions.objects)
-        elif hasattr(predictions, "segmentedObjects"):
-            detections = _detection2array(predictions.segmentedObjects)
```

Comment:
Looks like either IsegTiler or Iseg result should be updated if any changes here

Reply:
Both IsegTiler or Iseg result refactored.